### ISSUE TITLE
[app.config.ts] Could not resolve "virtual:app-config"

### ISSUE THREAD

**Issue by aklinker1** [wxt-dev/wxt#830]
### Discussed in https://github.com/wxt-dev/wxt/discussions/828

<div type='discussions-op-text'>

<sup>Originally posted by **devjume** July 19, 2024</sup>
Hi,

I am wondering what is correct way of using `app.config.ts` in wxt?
I am encountering error when using it in svelte component, but there is no error when using it in `background.ts`.

**Problem:**
```
// test.svelte - Doesn't work
<script lang="ts">
  import { useAppConfig } from 'wxt/client';
  console.log(useAppConfig());
</script>
```
Previous svelte component causes error:
```
Could not resolve "virtual:app-config"
node_modules/.pnpm/wxt@0.18.14_@types+node@20.14.11_rollup@4.18.1/node_modules/wxt/dist/client.js:414:22:
414 │ import appConfig from "virtual:app-config";
 You can mark the path "virtual:app-config" as external to exclude it from the bundle, which will
  remove this error and leave the unresolved path in the bundle.
```
When I use it in `background.ts` it works just as stated in documentation.
```
// background.ts - Works
import { useAppConfig } from 'wxt/client';
console.log(useAppConfig());
```
Of course I can import `defineAppConfig` object in svelte as follows, which works. But is that correct way?
```
<script lang="ts">
  import appConfig from '@/app.config';
  console.log("appConfig", appConfig);
</script>
```
I am not familiar how `app.config.ts` is used in Nuxt.  And Wxt docs didn't give any specific notes how to use it. Other than `console.log(useAppConfig());`.
wxt version: `0.18.14`

</div>

**Comment by aklinker1** [comment]
OK, looks like the solution is to add the following vite config:

```ts
  vite: () => ({
    optimizeDeps: {
      exclude: ["virtual:app-config"],
    },
  }),
```

This tells ESBuild that the `virtual:app-config` module is external, and that esbuild shouldn't try and resolve the value. Then when vite tries to resolve it, the [custom vite plugin used to resolve virtual modules](https://github.com/wxt-dev/wxt/blob/7ee391f6881b401cde3a46eea94bc4e6fb9ba0ab/packages/wxt/src/core/builders/vite/plugins/resolveAppConfig.ts#L16-L25) finds the correct file.

**Comment by aklinker1** [comment]
Released in [`v0.18.15`](https://github.com/wxt-dev/wxt/releases/tag/wxt-v0.18.15)
